scripts/migration/initialize_menas.py

```diff
@@ -0,0 +1,167 @@
+
+    # either version=1 record exists, or create new
+    if version_record:
+        if version_record["version"] != 1:
+            # failing on incompatible version, because that may corrupt data
+            raise MenasDbVersionError(f"Existing incompatible version record has been found in"
+                                      f" {DB_VERSION_COLLECTION}{self.hint}: {version_record}")
+        else:
+            print("  Existing db_version version=1 record found.")
+    else:
+        insert_result = collection.insert_one({"version": 1})
+        print(f"  Created version=1 record with id {insert_result.inserted_id}")
+    print("")
+
+
+def create_collections_if_not_exist(db: Database, collection_names: List[str]):
+    print("Initialization of migration-related collections")
+    existing_collections = db.list_collection_names()
+    if verbose:
+        print(f"  BEFORE: Aiming to create {collection_names}, found exising {existing_collections}.")
+
+    left_to_create = list(filter(lambda col_name: not(col_name in existing_collections), collection_names))
+    print(f"  Creating missing collections {left_to_create} ...")
+    for missing_col in left_to_create:
+        db.create_collection(missing_col)
+
+    if verbose:
+        print(f"  AFTER: existing collections: {db.list_collection_names()}")
+    print("")
+
+
+def initialize_collection_indices(db: Database, name: str, indices: List[dict]):
+    col = db[name]
+    if verbose:
+        print(f"  BEFORE: index info: {col.index_information()}")
+
+    # Example index info:
+    # index info: {'_id_': {'v': 2, 'key': [('_id', 1)], 'ns': 'menas.run_v1'}, 'dataset_1_datasetVersion_1_runId_1':
+    # {'v': 2, 'unique': True, 'key': [('dataset', 1), ('datasetVersion', 1), ('runId', 1)], 'ns': 'menas.run_v1'}}
+
+    # existing indices = no-op
+    for desired_index in indices:
+        key = desired_index[INDICES_FIELD_KEY]
+        unique = desired_index[INDICES_FIELD_UNIQUE]
+        print(f"  creating index for key {key}, unique={unique}")
+        col.create_index(key, unique=unique)
+
+    if verbose:
+        print(f"  AFTER: index info: {col.index_information()}")
+
+
+def run(parsed_args: argparse.Namespace):
+    target = parsed_args.target
+    target_db_name = parsed_args.targetdb
+
+    dryrun = args.dryrun  # if set, only migration description will be printed, no actual migration will run
+
+    print('Menas mongo initialization')
+    print('Running with settings: dryrun={}, verbose={}'.format(dryrun, verbose))
+    print('  target connection-string: {}'.format(target))
+    print('  target DB: {}'.format(target_db_name))
+    print("")
+
+    from menas_db import MenasDbError, MenasDb
+    target_db: MenasDb = MenasDb.from_connection_string(target, target_db_name, "target db", verbose)
+
+    try:
+        target_db.check_db_version()
+        target_db.check_menas_collections_exist()  # todo, check indexing, too?
```

Comment:
I would not be so nice. You implemented create or repair basically. I do not think that should be the job of this script.

This script should check db_version collection. And if it is:
- none -> Start init
- < 1 -> Tell user to run means for upgrade
- == 1 -> Tell user all is fine

Reply:
Redone in this fashion, thanks for the real-world-inspired opinion.